addons/mail/static/tests/owl/component/activity_tests.js

```diff
@@ -0,0 +1,832 @@
+    const fiveDaysBeforeNow = new Date();
+    fiveDaysBeforeNow.setDate(today.getDate() - 5);
+    const activityLocalId = this.env.store.dispatch('_createActivity', {
+        date_deadline: date_to_str(fiveDaysBeforeNow),
+        state: 'overdue',
+    });
+    await this.createActivityComponent(activityLocalId);
+    assert.strictEqual(
+        document.querySelectorAll('.o_Activity').length,
+        1,
+        "should have activity component in DOM"
+    );
+    const activity = document.querySelector('.o_Activity');
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_dueDateText').length,
+        1,
+        "should have activity delay in DOM"
+    );
+    assert.ok(
+        activity.querySelector('.o_Activity_dueDateText').classList.contains('o-overdue'),
+        "activity delay should have the right color modifier class (overdue)"
+    );
+    assert.strictEqual(
+        activity.querySelector('.o_Activity_dueDateText').textContent,
+        "5 days overdue:",
+        "activity delay should have '5 days overdue:' as label"
+    );
+});
+
+QUnit.test('activity with a summary layout', async function (assert) {
+    assert.expect(4);
+
+    await this.start();
+    const activityLocalId = this.env.store.dispatch('_createActivity', {
+        summary: 'test summary',
+    });
+    await this.createActivityComponent(activityLocalId);
+    assert.strictEqual(
+        document.querySelectorAll('.o_Activity').length,
+        1,
+        "should have activity component in DOM"
+    );
+    const activity = document.querySelector('.o_Activity');
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_summary').length,
+        1,
+        "should have activity summary"
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_type').length,
+        0,
+        "should not have activity type"
+    );
+    assert.strictEqual(
+        activity.querySelector('.o_Activity_summary').textContent.trim(),
+        "“test summary”",
+        "should have the activity summary in activity summary div"
+    );
+});
+
+QUnit.test('activity without summary layout', async function (assert) {
+    assert.expect(5);
+
+    await this.start();
+    const activityLocalId = this.env.store.dispatch('_createActivity', {
+        activity_type_id: [1, 'Fake type'],
+    });
+    await this.createActivityComponent(activityLocalId);
+    assert.strictEqual(
+        document.querySelectorAll('.o_Activity').length,
+        1,
+        "should have activity component in DOM"
+    );
+    const activity = document.querySelector('.o_Activity');
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_type').length,
+        1,
+        "should have activity sidebar type"
+    );
+    assert.strictEqual(
+        activity.querySelector('.o_Activity_type').textContent.trim(),
+        "Fake type",
+        "should have the activity type display name in activity type div"
```

Comment:
`div` is an implementation detail, assertion doesn't care of that (could be any other tag)